templates/human.html

```diff
@@ -210,18 +210,18 @@ <h4>Human behavior configuration:</h4>
         extra = renderExtra(extra, platform);
         switch (platform) {
             case "darwin":
-                baseHuman = 'curl -sk -o '+humanName+'.tar.gz -X POST -H \'file:'+humanName+'.tar.gz\' '+http+'/file/download 2>&1 && mkdir '+humanName+' && tar -C '+humanName+' -zxvf '+humanName+'.tar.gz ' +
-                    '&& cd '+humanName+' && virtualenv -p python3 '+humanName+' && '+humanName+'/bin/pip install -r requirements.txt && '+humanName+'/bin/python human.py --clustersize '+taskCount+' ' +
+                baseHuman = 'curl -sk -o \''+humanName+'.tar.gz\' -X POST -H \'file:'+humanName+'.tar.gz\' '+http+'/file/download 2>&1 && mkdir \''+humanName+'\' && tar -C \''+humanName+'\' -zxvf \''+humanName+'.tar.gz\' ' +
+                    '&& cd \''+humanName+'\' && virtualenv -p python3 \''+humanName+'\' && \''+humanName+'/bin/pip\' install -r requirements.txt && \''+humanName+'/bin/python\' human.py --clustersize '+taskCount+' ' +
                     '--taskinterval '+taskInterval+' --taskgroupinterval '+taskClusterInterval+' --extra '+extra;
                 break;
             case "linux":
-                baseHuman = 'curl -sk -o '+humanName+'.tar.gz -X POST -H \'file:'+humanName+'.tar.gz\' '+http+'/file/download 2>&1 && mkdir '+humanName+' && tar -C '+humanName+' -zxvf '+humanName+'.tar.gz ' +
-                    '&& cd '+humanName+' && virtualenv -p python3 '+humanName+' && '+humanName+'/bin/pip install -r requirements.txt && '+humanName+'/bin/python human.py --clustersize '+taskCount+' ' +
+                baseHuman = 'curl -sk -o \''+humanName+'.tar.gz\' -X POST -H \'file:'+humanName+'.tar.gz\' '+http+'/file/download 2>&1 && mkdir \''+humanName+'\' && tar -C \''+humanName+'\' -zxvf \''+humanName+'.tar.gz\' ' +
+                    '&& cd \''+humanName+'\' && virtualenv -p python3 \''+humanName+'\' && \''+humanName+'/bin/pip\' install -r requirements.txt && \''+humanName+'/bin/python\' human.py --clustersize '+taskCount+' ' +
                     '--taskinterval '+taskInterval+' --taskgroupinterval '+taskClusterInterval+' --extra '+extra;
                 break;
             case "windows-psh":
                 baseHuman = '$server="'+http+'"; $url="$server/file/download"; $wc=New-Object System.Net.WebClient; $wc.Headers.add("file","'+humanName+'.zip"); $wc.DownloadFile($url, "$pwd\\'+humanName+'.zip"); ' +
-                    'Expand-Archive '+humanName+'.zip -DestinationPath '+humanName+' -Force; cd '+humanName+'; python.exe -m venv '+humanName+';' +
+                    'Expand-Archive "'+humanName+'.zip" -DestinationPath "'+humanName+'" -Force; cd "'+humanName+'"; python.exe -m venv "'+humanName+'";' +
```

Comment:
Any idea why we use python.exe here but the python.exe packaged within the human zip (in `Scripts`) in the next line? Perhaps for consistency but also to possibly avoid requiring the installation of python on the workstation we should use the python.exe inside the `Scripts` directory for both. Worth testing to ensure we can get around that implicit requirement.

Reply:
Resolving since we need python to create the virtual environment in the first place, good catch!